packages/aws-cdk-lib/aws-codepipeline-actions/test/codestar-connections/trigger.test.ts

```diff
@@ -0,0 +1,236 @@
+import { Filter, PushFilter, Trigger } from '../../lib/codestar-connections/trigger';
+
+describe('CodeStarSourceConnections Trigger Tests', () => {
+  describe('Filter tests', () => {
+    describe('Filter on push', () => {
+      test('onTags returns expected filters when both includes and excludes are provided', () => {
+        const filter = Filter.push(PushFilter.onTags({
+          includes: ['tag1', 'tag2'],
+          excludes: ['notThisTag', 'i-dont-know-her'],
+        }));
+
+        expect(filter._push?.tags).toEqual({ includes: ['tag1', 'tag2'], excludes: ['notThisTag', 'i-dont-know-her'] });
+        expect(filter._push?.branches).toBeUndefined();
+        expect(filter._push?.filePaths).toBeUndefined();
+        expect(filter._pullRequest).toBeUndefined();
+      });
+
+      test('onTags return expected filters when only includes is provided', () => {
+        const filter = Filter.push(PushFilter.onTags({
+          includes: ['tag1', 'tag2'],
+        }));
+
+        expect(filter._push?.tags).toEqual({ includes: ['tag1', 'tag2'] });
+        expect(filter._push?.branches).toBeUndefined();
+        expect(filter._push?.filePaths).toBeUndefined();
+        expect(filter._pullRequest).toBeUndefined();
```

Comment:
This looks like testing implementation details, but I appreciate the brevity of these tests. We should have a unit test that shows using a Filter with a Trigger on an Action, so we get some tests that cover (and show) the intended usage.